### PR TITLE
Add ability to specify extra pod labels in helm chart

### LINKED ISSUE #566: Add ability to specify extra pod labels in helm chart
**Is your feature request related to a problem? Please describe.**
I am unable to specify additional labels on the pods created by the botkube helm chart.  This would be useful functionality to have.

**Describe the solution you'd like**
Allow me to specify `extraLabels` similar to `extraAnnotations` in the helm chart.  

**Additional context**
We use datadog.  Datadog likes us to provide [specific labels](https://docs.datadoghq.com/getting_started/tagging/unified_service_tagging/?tab=kubernetes) so it can identify pods and assign them to its understanding of services.

### PR DESCRIPTION
**Pull request by ezodude**
## Description

This PR allows users to update helm chart config with extra labels.

These labels are then assigned to created BotKube Pods.

Changes proposed in this pull request:

- New `extraLabels` config field.
- Update to the helm `deployment.yaml` template to apply any `extraLabels` that are found.
- Single target builds to get around building images on M1 with `qemu`.
- Updated contributing documentation.

## Related issue(s)

Resolves #566

### CODE REVIEW

**Comment by ezodude** [comment]
Tested locally and it works, but having issues with `make container-image` and `hack/goreleaser.sh` ... specifically:

```
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
Error while loading /qemu-binfmt-conf.sh: Exec format error
make: *** [container-image] Error 1
```

I'm investigating the issue. [I need this to work to test it that it work on my local K8s](https://github.com/kubeshop/botkube/blob/main/CONTRIBUTING.md#build-and-install-on-kubernetes).

**Comment by pkosiec** [comment]
@ezodude In this PR there are purely Helm chart changes, which means it's not necessary to rebuild the image, as it wasn't changed in any way 🙂 But anyway - true, this issue needs to be investigated for future (look for some threads around M1, as it is definitely related to it). 🤞

**Comment by ezodude** [comment]
Aborting trying to fix the [arm64 build issue I was having earlier](https://github.com/kubeshop/botkube/pull/661#issuecomment-1194348892).

Especially, given this is just a helm chart change (with no required code updates) I think it’s ready for review.

**Comment by ezodude** [comment]
@pkosiec @mszostok the last commit sorts out dev builds. 

- Works around `qemu` issues completely.
- I've tested it locally end to end and it works. 
- Here's the [created image on Docker Hub](https://hub.docker.com/r/ezokubeshop/botkube/tags).

Would love to hear your thoughts.

(Back to draft status as I need to update the docs for local dev to use `make container-image-single`).

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -31,15 +31,41 @@ This section describes how to build and run the BotKube from the source code.
...
+**Multi-arch target builds for any K8s cluster**
+
+This is ideal for running BotKub on remote clusters. 
+
+When tagging your dev image _take care to add your target image architecture as a suffix_. For example, in the command below we added `-amd64` as our target architecture.
+
+This ensures the image will run correctly on the target K8s cluster. 
+
+Note: This command takes sometime to run as it builds for multiple architectures.  
+
```
Comment (mszostok) [inline CONTRIBUTING.md:60]: a few **nit** suggestions:
- missing indent
- "MiniKube" - the `minikube` is just all lower case
- "As a side benefit, it gets around some issues of building BotKube on M1 Chips." - IMO this note is not needed. 
- "Note: " - we use a GitHub "native" note.

BTW, I like a lot that you added "This is ideal (..)" as it gives a nice context 👍 

```suggestion
1. Build BotKube and create a new container image tagged as `ghcr.io/kubeshop/botkube:v9.99.9-dev`. Choose one option:

   - **Single target build for your local K8s cluster**

     This is ideal for running BotKube on a local cluster, e.g. using [kind](https://kind.sigs.k8s.io) or [`minikube`](https://minikube.sigs.k8s.io/docs/).

     Remember to set the `IMAGE_PLATFORM` env var to your target architecture. For example, the command below builds the `linux/arm64` target. By default, the build targets `linux/amd64`.

        ```sh
        IMAGE_PLATFORM=linux/arm64 make container-image-single
        docker tag ghcr.io/kubeshop/botkube:v9.99.9-dev <your_account>/botkube:v9.99.9-dev
        docker push <your_account>/botkube:v9.99.9-dev
        ```
        Where `<your_account>` is Docker hub account to which you can push the image.

   - **Multi-arch target builds for any K8s cluster**

     This is ideal for running BotKube on remote clusters.

     When tagging your dev image take care to add your target image architecture as a suffix. For example, in the command below we added `-amd64` as our target architecture.

     This ensures the image will run correctly on the target K8s cluster.

     > **Note**
     > This command takes some time to run as it builds the images for multiple architectures.

```

Renders as:
![Screen Shot 2022-07-27 at 15 05 43](https://user-images.githubusercontent.com/17568639/181255133-f91d5922-cf7e-4e31-9a4c-ea8525515f84.png)

File: hack/goreleaser.sh [hunk 2]
```diff
@@ -98,6 +99,19 @@ build() {
...
+build_single() {
+  export GORELEASER_CURRENT_TAG=v9.99.9-dev
+  docker run --rm --privileged \
+    -v "$PWD":/go/src/github.com/kubeshop/botkube \
+    -v /var/run/docker.sock:/var/run/docker.sock \
+    -w /go/src/github.com/kubeshop/botkube \
+    -e GORELEASER_CURRENT_TAG=${GORELEASER_CURRENT_TAG} \
+    -e ANALYTICS_API_KEY="${ANALYTICS_API_KEY}" \
+    goreleaser/goreleaser build --single-target --rm-dist --snapshot --id botkube -o "./botkube"
+  docker build --no-cache -f "$PWD/build/Dockerfile" --platform "${IMAGE_PLATFORM}" -t "${IMAGE_REPOSITORY}:${GORELEASER_CURRENT_TAG}" .
```
Comment (mszostok) [inline hack/goreleaser.sh:111]: why `--no-cache` is used?

**Review by mszostok — APPROVED** [review]
In general, LGTM 🚀 

I tested it with:
```bash
helm template ./helm/botkube --set extraLabels.foo=bar,extraLabels.test=baz --show-only templates/deployment.yaml
```

Please also port CONTRIBUTION.md related changes to the `botkub-docs` (https://github.com/kubeshop/botkube-docs/blob/main/content/contribute/_index.md). For now, it's not automated. We also should rethink if we want to have it in both places, maybe a reference will be better.

File: hack/goreleaser.sh (hunk 2 above)
Comment (ezodude) [reply hack/goreleaser.sh:111]: Copy and paste. Will fix.

**Pull request merged**